### DISCUSSION TITLE
Does ezdxf support compression of DXF files？

### DISCUSSION THREAD

**Discussion by wangm23456** [mozman/ezdxf#531] [Q&A]
Thank you very much for your work.
Does eZDXF support compression of DXF, such as Zlib like Klayout？

**Comment by wangm23456** [comment]
Through KLayout's conversion, I can compress the DXF file 30 times, can I implemented in EZDXF?
For example, add compression feature in Drawing.saveas() function.

![image](https://user-images.githubusercontent.com/13377123/132220506-9e43a24c-c212-45df-b990-1236cd3f803d.png)

**Reply by mozman** [reply]
Has the "compressed" version the same content as the original DXF file?

**Reply by wangm23456** [reply]
yes,  it's the same.
I set  "zlib compression" and "write hatch entity".

![image](https://user-images.githubusercontent.com/13377123/132294738-5a8061c1-385e-4a81-b9f9-10165fb2b523.png)

**Reply by wangm23456** [reply]
the compressed dxf can be unzip as:
![image](https://user-images.githubusercontent.com/13377123/132295898-34a8dee7-d774-4956-91b9-c8ef4e8e7b12.png)

the compressed dxf can be read directly by klayout.

**Reply by mozman** [reply]
> the compressed dxf can be read directly by klayout.

... but not by any other CAD application (AutoCAD, BricsCAD, ...) see my post below and using the file extension `.dxf` for zipped DXF files is just wrong.

**Reply by wangm23456** [reply]
Yes, I still don't know how klayout does it, or maybe it's not the common practice .

**Reply by mozman** [reply]
Your screenshot above says: packing the DXF file into a archive by using the zlib compression algorithm and saving the zip file with the incorrect extension `.dxf` instead of `.zip`

**Comment by mozman** [comment]
The DXF format has no builtin compression feature, but you can always pack DXF documents into zip-files. These packed DXF files have to be unpacked before they can be open by a CAD application or viewer and `ezdxf` can read DXF files direct from zip-files (see [docs](https://ezdxf.mozman.at/docs/drawing/management.html#ezdxf.readzip)). To store DXF files in zip-files use the Python standard [zipfile](https://docs.python.org/3/library/zipfile.html) module.
